VERSIONING.md

```diff
@@ -0,0 +1,244 @@
+(Ideally we need an "Experimental" attribute, but .NET has only built-in support
+for "Obsolete", so we are leveraging it. Using "Obsolete" code results in
+compile time warnings to caution users.) This allows an experimental signal to
+co-exist in the same package as other, non-experimental signals. For example,
+the OpenTelemetry 1.0.0 package will consist of Traces and Metrics. As Metrics
+signal is not ready for stable release, every method dealing with Metrics will
+be marked "Obsolete". Once Metrics signal gets stable, it'll be added to a
+release with minor version bump, say 1.2.0. The "Obsolete" Metric methods can be
+removed later with a major version bump.
+
+### Downsides with Approach 1
+
+It could be confusing if the term `Obsolete` is used for more than one purpose -
+one is the original intended usage for marking Obsolete things, and the
+additional usage for experimental signals. It *may* be used for Metrics, as the
+current Metrics API is based on a draft version of the spec which is already
+removed. As we are building a new metrics API, we need to deprecate the original
+ones, so marking them as `Obsolete` is justified. However, this approach cannot
+be used for the general purpose of shipping experimental features.
+
+### Example with Approach 1
+
+Following shows an example on how the `OpenTelemetry` package versioning works
+with this approach:
+
+`OpenTelemetry` 0.7.0-beta1 release : Pre-release, no API guarantees.
+
+`OpenTelemetry` 1.0.0-RC1 release : Pre-release, no API guarantees, but more
+stable than beta.
+
+`OpenTelemetry` 1.0.0 release : Stable release consisting of Traces, Propagators
+and Metrics. Metrics will be marked "Obsolete", to warn users that Metrics is
+not Stable.
+
+`OpenTelemetry` 1.0.1 release : Bug fixes.
+
+`OpenTelemetry` 1.1.0 release : New features added.
+
+`OpenTelemetry` 1.2.0 release : Add stable metric support. This will be additive
+changes. Any existing Obsolete Metric code will continue to be present.
+
+`OpenTelemetry` 1.3.0 release : More features.
+
+`OpenTelemetry` 2.0.0 release : Remove all Obsolete methods from 1.* releases.
+
+### Approach 2 - Separate package for experimental
+
+An alternate option to deal with experimental signal is to remove the signal
+from the common packages, and ship as separate package (i.e a package with a
+different name.) For example, OpenTelemetry 1.0.0 will only contain stable
+signals like Traces, Baggage, Propagators. There will be a separate packages
+like OpenTelemetry.API.Metrics and OpenTelemetry.Metrics, which contains the
+metric API and SDK respectively. The packages can be released as pre-releases
+with no stability guarantees. Once the signal achieve stable quality, it'll be
+made part of the main package and released as a minor version update.
+
+### Downsides with Approach 2
+
+The number of packages will explode over time. There is no option to delete a
+package from nuget.org, so the experimental packages will remain as orphan ones,
```

Comment:
There are ways of reducing this confusion. Packages cannot be removed but they can be unlisted. The can also be marked deprecated with a pointer to the package that replaces it.

Reply:
ya its an option. (delisting is manual work in nuget.org)